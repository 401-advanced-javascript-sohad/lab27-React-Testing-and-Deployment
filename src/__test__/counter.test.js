import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import Counter from '../components/counter/main.js';


describe("<Counter />", () => {
 
  it('is alive in our application', () => {
    let app = shallow(<Counter />);
    expect(app.find('span').exists()).toBeFalsy();
  });



  

  it("negative --> changes state on a click", () => {
    let component = mount(<Counter />);
    let button = component.find("#clickOne");
    button.simulate("click");

    expect(component.state("counter")).toBe(-1);
  });

  it("positive --> changes state on a click", () => {
    let component = mount(<Counter />);
    let button = component.find("#clickTwo");
    button.simulate("click");

    expect(component.state("counter")).toBe(1);
  });



 
  it("renders correctly", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});