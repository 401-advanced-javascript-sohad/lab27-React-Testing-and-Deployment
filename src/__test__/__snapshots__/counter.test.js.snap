// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section>
  <h3>
    0
  </h3>
  <button
    onClick={[Function]}
  >
    click for Increment
  </button>
  <button
    onClick={[Function]}
  >
    click for Decrement
  </button>
</section>
`;
